### ISSUE TITLE
Shell heredocs not supported?

### ISSUE THREAD

**Issue by madrobby** [ansible/ansible#6511]
##### Issue Type:

Feature Idea
##### Ansible Version:

ansible 1.5.2 (release1.5.3 48a1c156ee) last updated 2014/03/13 14:03:23 (GMT -400)
##### Environment:

OS X -> Ubuntu 12.04
##### Summary:

It would be great if Ansible would support shell heredocs:

``` sh
shell: >
    psql -d freckle <<EOF
    CREATE USER freckle_owner WITH PASSWORD '{{ postgresql_freckle_owner_password }}';
    CREATE USER freckle_app WITH PASSWORD '{{ postgresql_freckle_app_password }}';
    ALTER DATABASE freckle OWNER TO freckle_owner;
    GRANT ALL PRIVILEGES ON DATABASE freckle TO freckle_owner;
    -- MORE SQL HERE
    EOF
```

Perhaps there's a workaround to make this run correctly, like calling bash directly? Or, is there some other way to make a multiple line input to `psql`? I don't want to use templates files because I don't want the passwords saved to disk on the remote server.
##### Steps To Reproduce:

See above.
##### Expected Results:

It should work. :)
##### Actual Results:

Fails as the lines are parsed as one shell command each.

**Comment by madrobby** [comment]
Setting `executable=/bin/bash` also fails, with this error:

``` sh
failed: [192.168.33.1] => {"changed": true, "cmd": " psql -d freckle <<EOF CREATE USER freckle_owner WITH PASSWORD 'foo'; CREATE USER freckle_app WITH PASSWORD 'bar'; ALTER DATABASE freckle OWNER TO freckle_owner; --etc etc etc\nEOF ", "delta": "0:00:00.052084", "end": "2014-03-16 13:43:37.618503", "item": "", "rc": 127, "start": "2014-03-16 13:43:37.566419"}
stderr: /bin/bash: line 4: warning: here-document at line 0 delimited by end-of-file (wanted `EOF')
```

**Comment by bcoca** [comment]
try using | instead of > , which preserves multiline input.

## 

Brian Coca
Stultorum infinitus est numerus
0110000101110010011001010110111000100111011101000010000001111001011011110111010100100000011100110110110101100001011100100111010000100001
Pedo mellon a minno

**Comment by madrobby** [comment]
@bcoca Thanks, will give that a try. I assumed that it's likely PEBKAC at fault here. :) Being new to Ansible I find the documentation lacking in more advanced examples. :/

**Comment by mpdehaan** [comment]
The shell module transfers a file and then runs it with arguments, the above is not meant to be a thing, which is why it is not shown in the examples.

**Comment by madrobby** [comment]
@mpdehaan I don't understand your answer, would you mind explaining in more detail?

**Comment by bcoca** [comment]
basically, shell is for one liners, if you want to run a script use the
script: module.

herdocs in shell/command/etc while might work, are undesired (makes for
very ugly playbooks).

i would argue they make for very ugly <language that supports heredocs>.

**Comment by madrobby** [comment]
It doesn't look like that the `script` module supports variables. I think heredocs are an elegant way to include little things like this where it would be awkward and very verbose to write lots of actions. :/

**Comment by AnneTheAgile** [comment]
@madrobby , the script module supports arguments, is that what you mean? 
http://docs.ansible.com/script_module.html
